### PR TITLE
Use Python 3.11

### LINKED ISSUE #75: Use GHCR for test image
at the moment its docker hub and GHCR is supposed to be faster with GH

### PR DESCRIPTION
**Pull request by jensens**
- Use Python 3.11
- Cleanup dependencies (Debian, pip), partially addresses #89 
- fixes #75

### CODE REVIEW

**Review by Rudd-O — APPROVED** [review]
Extra twin turbo hotness.

**Comment by jensens** [comment]
The possible problem is, there are some packages with C-extensions in the Zope world pulled in and the image does not provide a build environment by default.

**Comment by jensens** [comment]
tests are passing, it is not working anyway.

File: Dockerfile [hunk 1]
```diff
@@ -37,8 +37,9 @@ WORKDIR /app
 COPY --from=builder --chown=500:500 /app /app
 
 RUN <<EOT
+    ls -la /app/bin
     useradd --system -m -d /app -U -u 500 plone
-    runDeps="git libjpeg62 libopenjp2-7 libpq5 libtiff5 libxml2 libxslt1.1 lynx netcat poppler-utils rsync wv busybox gosu libmagic1 make"
+    runDeps="git libjpeg62 libopenjp2-7 libpq5 libtiff5 libxml2 libxslt1.1 lynx netcat poppler-utils rsync wv busybox gosu libmagic1 build-essential"
```
Comment (ericof) [inline Dockerfile]: Why do you add build-essential here? Do we need all of it in the final image?

File: Dockerfile [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
 ENV PIP_PARAMS=""
-ENV PIP_VERSION=22.2.2
+ENV PIP_VERSION=22.3.1
 ENV PLONE_VERSION=6.0.0rc1
 ENV EXTRA_PACKAGES="relstorage==3.4.5 psycopg2==2.9.3 python-ldap==3.4.0"
 
 RUN <<EOT
     apt-get update
-    buildDeps="dpkg-dev gcc libbz2-dev libc6-dev libffi-dev libjpeg62-turbo-dev libldap2-dev libopenjp2-7-dev libpcre3-dev libpq-dev libsasl2-dev libssl-dev libtiff5-dev libxml2-dev libxslt1-dev wget zlib1g-dev python3-dev build-essential"
+    buildDeps="build-essential libbz2-dev libffi-dev libjpeg62-turbo-dev libldap2-dev libopenjp2-7-dev libpcre3-dev libpq-dev libsasl2-dev libssl-dev libtiff5-dev libxml2-dev libxslt1-dev wget zlib1g-dev"
```
Comment (ericof) [inline Dockerfile:13]: Don't we need python3-dev?

**Review by ericof — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (jensens) [reply Dockerfile:13]: No. The base image `python-dev` is on 3.9. The 3.11 is in `/usr/local/*` and includes the dev-headers for 3.11.

File: Dockerfile (hunk 1 above)
Comment (jensens) [reply Dockerfile]: Probably not need, its a leftover from my trials. I'll have a look.

Comment (jensens) [reply Dockerfile]: OK, I removed it.

**Review by davisagli — COMMENTED** [review]
Having an image with Python 3.11 sounds good, but it is also a significant change. There may be addons or project code that is not compatible with it yet. Do we need to consider publishing images for multiple versions of Python?

**Comment by Rudd-O** [comment]
> Having an image with Python 3.11 sounds good, but it is also a significant change. There may be addons or project code that is not compatible with it yet. Do we need to consider publishing images for multiple versions of Python?

I think the risk that code which worked with Python 3.9 and won't work with Python 3.11 is fairly low, and worth the benefits.

**Comment by jensens** [comment]
Two more thoughts: 
- performance is much better
- green IT -> lower CO2 footprint

**Comment by davisagli** [comment]
I won't block this (I want to try Python 3.11 also). But I do think there is a nonzero chance we'll need to start building an image with Python 3.10 as well, for example if there are addons which are not compatible with Python 3.10 yet.

**Comment by jensens** [comment]
> But I do think there is a nonzero chance we'll need to start building an image with Python 3.10 as well, for example if there are addons which are not compatible with Python 3.10 yet.

I think this is fine, but I would wait for complaints instead of providing one proactive. I go for merge.

**Pull request merged**